packages/webapp/src/routes/__tests__/developer/confirm-off-site-gain.spec.js

```diff
@@ -0,0 +1,284 @@
+          {
+            'Hedgerow type': 'Native hedgerow - associated with bank or ditch',
+            'Length (km)': 3,
+            'Total hedgerow units': 27,
+            Condition: 'Good'
+          }
+        ]
+      }
+
+      const mockContextResult = {
+        gainSiteNumber: undefined,
+        offSiteHabitats: {
+          items: [
+            {
+              'Area (hectares)': 1,
+              'Broad habitat': 'Rocky shore ',
+              Condition: 'Fairly Good',
+              'Habitat type': 'Moderate energy littoral rock - on peat, clay or chalk',
+              'Total habitat units': 0
+            }
+          ],
+          total: 1
+        },
+        offSiteHedgerows: {
+          items: [
+            {
+              Condition: 'Good',
+              'Hedgerow type': 'Native hedgerow - associated with bank or ditch',
+              'Length (km)': 3,
+              'Total hedgerow units': 27
+            }
+          ],
+          total: 3
+        }
+      }
+      redisMap.set(constants.redisKeys.DEVELOPER_METRIC_DATA, _mockMetricData)
+
+      const request = {
+        yar: redisMap
+      }
+      const h = {
+        view: (view, context) => {
+          viewResult = view
+          contextResult = context
+        }
+      }
+      await confirmOffsiteGainOptions.default[0].handler(request, h)
+      expect(viewResult).toEqual(constants.views.DEVELOPER_CONFIRM_OFF_SITE_GAIN)
+      expect(contextResult).toMatchObject(mockContextResult)
+    })
+
+    it(`should render the ${url.substring(1)} view with some insufficient data`, async () => {
+      const confirmOffsiteGainOptions = require('../../developer/confirm-off-site-gain.js')
+      const _mockMetricData = {
+        d1OffSiteHabitatBaseline: [
+          {
+            'Area (hectares)': 1,
+            'Total habitat units': 0
+          }
+        ],
+        e1OffSiteHedgeBaseline: [
+          {
+            'Length (km)': 3
+          }
+        ]
+      }
+      const mockContextResult = {
+        gainSiteNumber: undefined,
+        offSiteHabitats: {
+          items: [{
+            'Area (hectares)': 1,
+            'Total habitat units': 0
+          }],
+          total: 0
+        },
+        offSiteHedgerows: {
+          items: [{
+            'Length (km)': 3
+          }],
+          total: 0
+        }
+      }
+      redisMap.set(constants.redisKeys.DEVELOPER_METRIC_DATA, _mockMetricData)
+
+      const request = {
+        yar: redisMap
+      }
+      const h = {
+        view: (view, context) => {
+          viewResult = view
+          contextResult = context
+        }
+      }
+      await confirmOffsiteGainOptions.default[0].handler(request, h)
+      expect(viewResult).toEqual(constants.views.DEVELOPER_CONFIRM_OFF_SITE_GAIN)
+      expect(contextResult).toMatchObject(mockContextResult)
+      expect(getNumOfUnits(mockMetricData.offSiteHabitatBaseline, 'Broad habitat', 'Area (hectares)')).toEqual(1)
```

Comment:
again testing the mock object rather than result, think these just need removing